src/bidiMapper/utils/mapperTabPage.ts

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright 2022 Google LLC.
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { LogType } from '../../utils/log';
+
+export class MapperTabPage {
+  // HTML source code for the user-facing Mapper tab.
+  static #mapperPageSource =
+    '<!DOCTYPE html><html lang="en"><head><title>BiDi-CDP Mapper</title><style>body{font-family: Roboto, serif; font-size: 13px; color: #202124;}.log{padding: 12px; margin-top: 16px; font-family: Menlo, serif; font-size: 11px; line-height: 180%; background: #f1f3f4; border-radius: 4px; display: flex; flex-direction: column; gap: 10px;}.pre{overflow-wrap: break-word;}.card{margin: 60px auto; padding: 2px 0; max-width: 900px; box-shadow: 0 1px 4px rgba(0, 0, 0, 0.15), 0 1px 6px rgba(0, 0, 0, 0.2); border-radius: 8px;}.divider{height: 1px; background: #f0f0f0;}.item{padding: 16px 20px;}</style></head><div class="card"><div class="item"><h1>BiDi-CDP Mapper is controlling this tab</h1><p>Closing or reloading it will stop the BiDi process. <a target="_blank" title="BiDi-CDP Mapper GitHub Repository" href="https://github.com/GoogleChromeLabs/chromium-bidi">Details.</a></p></div><div class="divider"></div><details id="details"><summary class="item">Debug information</summary></details></div></html>';
```

Comment:
Since we’re [omitting optional `<body>`/`</body>` tags](https://google.github.io/styleguide/htmlcssguide.html#Optional_Tags), let’s do the same for `html` and `head`.

```suggestion
    '<!DOCTYPE html><html lang="en"><title>BiDi-CDP Mapper</title><style>body{font-family: Roboto, serif; font-size: 13px; color: #202124;}.log{padding: 12px; margin-top: 16px; font-family: Menlo, serif; font-size: 11px; line-height: 180%; background: #f1f3f4; border-radius: 4px; display: flex; flex-direction: column; gap: 10px;}.pre{overflow-wrap: break-word;}.card{margin: 60px auto; padding: 2px 0; max-width: 900px; box-shadow: 0 1px 4px rgba(0, 0, 0, 0.15), 0 1px 6px rgba(0, 0, 0, 0.2); border-radius: 8px;}.divider{height: 1px; background: #f0f0f0;}.item{padding: 16px 20px;}</style></head><div class="card"><div class="item"><h1>BiDi-CDP Mapper is controlling this tab</h1><p>Closing or reloading it will stop the BiDi process. <a target="_blank" title="BiDi-CDP Mapper GitHub Repository" href="https://github.com/GoogleChromeLabs/chromium-bidi">Details.</a></p></div><div class="divider"></div><details id="details"><summary class="item">Debug information</summary></details></div>';
```

Reply:
done